state/factory/workingset.go

```diff
@@ -520,6 +554,9 @@ func (ws *workingSet) ValidateBlock(ctx context.Context, blk *block.Block) error
 	if err := ws.validateNonce(ctx, blk); err != nil {
 		return errors.Wrap(err, "failed to validate nonce")
 	}
+	if err := ws.validateSystemActionLayout(ctx, blk.RunnableActions().Actions()); err != nil {
```

Comment:
for safety, this should be gated by a hard-fork
then after chain passed the hard-fork height, we can remove the gating check